app/controllers/api/v1/orders_controller.rb

```diff
@@ -131,6 +123,13 @@ def order_response(records)
           exclude_stockit_set_item: true).to_json
       end
 
+      def order_with_orders_count
+        Order.users_order_count(@order.created_by_id).each do |key, value|
```

Comment:
i think a better approach would be to replace the class scope with attribute readers on the order model. You're halfway there already.

Try this:
- delete `users_order_count`
- add the following on `order.rb`

```
def submitted_count
  Order.where(created_by_id: self.created_by_id, state: 'submitted').size
end
```

then reference `submitted_count` in the serializer. 
Once this is done, we should check the speed, it could still be quite slow for the index methods.

Another approach would be to send this data down as `meta` outside of the object. This would be much faster and not require calculating for every order.

So in the controller, `render json: { meta: { counts: {submitted: 2, cancelled: 4} } }.merge(OrderSerializer.new(@results).as_json)...`